test/testdata/infer/infer1.rb

```diff
@@ -32,7 +32,7 @@ def baz6(cond)
  else
    b = "foo"
  end
- b = "foo".getbyte(b) # error: Expected `Integer` but found `T.any(Integer, String)` for argument `arg0`
+ b = "foo".getbyte(b) # error: Expected `Integer` but found `T.any(String, Integer)` for argument `arg0`
```

Comment:
Do you have a sense of why this changed?

Reply:
I believe it is because we sort on `kind` somewhere, and because the PR added:

https://github.com/sorbet/sorbet/pull/5614/files#diff-8a69c6e75a3c53920d6c4a7e8b59337d5c99e35ed53d3b83b9fc0bb821726d00R117